pkg/services/apiserver/service.go

```diff
@@ -178,13 +181,19 @@ func ProvideService(
 			}
 
 			if c.SignedInUser != nil {
+				// For unauthenticated requests, we set the namespace to the requested one
+				if !c.IsSignedIn {
+					useNamespaceFromPath(req.URL.Path, c.SignedInUser)
+				}
+
 				ctx := identity.WithRequester(req.Context(), c.SignedInUser)
 				req = req.WithContext(ctx)
 			}
 
 			resp := responsewriter.WrapForHTTP1Or2(c.Resp)
 			s.handler.ServeHTTP(resp, req)
 		}
+		k8sRoute.Any("/features.grafana.app/v0alpha1/*", handler)
```

Comment:
👍🏻  this is an OK quick fix that will give 403 (rather than 401) when making Unauthenticated requests.

We don't want to apply this generally, but selectively is OK for now.

I will start the process of seeing if we can get 401 supported here: https://github.com/kubernetes/kubernetes/blob/v1.33.1/staging/src/k8s.io/apiserver/pkg/endpoints/filters/authorization.go#L88